sandstone/pattern-virtuallist-preserving-focus/src/views/PatternList.js

```diff
@@ -2,42 +2,38 @@ import Item from '@enact/sandstone/Item';
 import VirtualList from '@enact/sandstone/VirtualList';
 import ri from '@enact/ui/resolution';
 import PropTypes from 'prop-types';
-import {Component} from 'react';
+import {useCallback} from 'react';
 
 import css from './PatternList.module.less';
 
 const items = Array.from(new Array(1000)).map((n, i) => `Item  ${('00' + i).slice(-3)}`);
 
-class PatternList extends Component {
-	static propTypes = {
-		id: PropTypes.string,
-		onClick: PropTypes.func
-	};
-
-	renderItem = ({index, ...rest}) => (
-		<Item {...rest} onClick={this.props.onClick}>
+const PatternList = ({id, onClick, ...rest}) => {
+	// eslint-disable-next-line
+	const renderItem = useCallback(({index, ...rest}) => (
```

Comment:
how about remove the line 12 and below?
```suggestion
	const renderItem = useCallback(({index, ...restProps}) => (
```

Reply:
fixed it.